src/index.ts

```diff
@@ -0,0 +1,165 @@
+import * as crypto from 'crypto'
+import * as path from 'path'
+import { EventEmitter } from 'events'
+const createLogger = require('ilp-logger')
+const log = createLogger('ilp-plugin')
+require('source-map-support').install()
+
+interface FunctionWithVersion extends Function {
+  version?: number
+}
+export interface ConnectOptions {
+  timeout?: number
+}
+export type DataHandler = (data: Buffer) => Promise<Buffer>
+export type MoneyHandler = (amount: string) => Promise<void>
+export interface PluginInstance extends EventEmitter {
```

Comment:
I think this type should just be called Plugin

Reply:
I'm trying to get some consistency across all our "module loading" types so I copied the `StoreInstance` type name from `ilp-connector`.

I also want to avoid using very common names.

Maybe a better convention is `IlpPlugin` and `IlpStore` etc?

Reply:
I don't think having a common name is a problem here because a) you don't need to import this interface b) if you do you're going to use it for the main plugin abstraction and c) if it does conflict with something else you've called Plugin you can always rename it. IlpPlugin seems a little redundant because you're getting it from a module called ilp something so there isn't any ambiguity about the context